### ISSUE TITLE
Indicate when another admin has active upgrades

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#96328]
### What

For admin users where the site's upgrades are owned by a different admin, we should provide an indication that another admin owns the purchases.

### Why

Admins on a site attempt to renew or cancel an upgrade, but face confusion if the upgrade is owned by a different admin because their purchases and billing history appear empty:
![Image](https://github.com/user-attachments/assets/f917d4e7-4dec-422c-96b2-35cced7d1506)
![Image](https://github.com/user-attachments/assets/c5b44ef1-9a35-4704-bd40-1fd3fe706792)

For the admin user, this makes it seem like the site has no upgrades, and they need to contact support to figure it out. That's when we inform them that the purchases are owned by a different admin, but we can surface this information in their /purchases/ to alleviate this confusion and prevent them from contacting support for help with it.

A typical example of this confusion can be seen in #9025258-zen

### How

There are a few different ways to approach this. Two suggestions:

#### List the upgrade with the admin's username
We can list the upgrade under /purchases/ in the same way as it appears for the owner of the upgrade, but with the admin's username provided in the notice:
![Image](https://github.com/user-attachments/assets/36c8d5ce-e4dc-4d17-b12a-e68bbe671fae)

#### Provide a generic notice
Place a simple alert at the top of the upgrades list with generic messaging.
![Image](https://github.com/user-attachments/assets/acf6d270-acbe-4305-9ecb-256bedc70f26)

Text used:
> ⚠️ Upgrades for yourgroovydomain.com are owned by other site admins. The purchase owner can manage this purchase.

**Comment by Robertght** [comment]
I agree and I can confirm I've seen this countless times over the years. It should also help our users have clarity over their account and subscriptions.

@Automattic/shilling I'm flagging this for your team

**Comment by JessBoctor** [comment]
I reviewed the [pending PR](https://github.com/Automattic/wp-calypso/pull/97198). The errors I was seeing in earlier reviews seem to have been resolved. I added my approval to @sirbrillig's earlier review. This is all set to ship.
